prow/config/config.go

```diff
@@ -241,6 +241,13 @@ type Spyglass struct {
 	// If left empty, the link will be not be shown. Otherwise, a GCS path (with no
 	// prefix or scheme) will be appended to GCSBrowserPrefix and shown to the user.
 	GCSBrowserPrefix string `json:"gcs_browser_prefix,omitempty"`
+	// TestGridConfig is the path to the TestGrid config proto. If the path begins with
+	// "gs://" it is assumed to be a GCS reference, otherwise it is read from the local filesystem.
+	// If left blank, TestGrid links will not appear.
+	TestGridConfig string `json:"testgrid_config,omitempty"`
```

Comment:
One concern I have with this approach is that people can't really deploy their own testgrid right now. Also, this will require giving deck/spyglass credentials to our testgrid bucket (probably not a big deal).

We could potentially write this lookup endpoint into testgrid, and then just have the TestGridRoot

Reply:
I agree that this approach is not so great for users who can't run a testgrid (i.e. everyone who isn't us).

I'm not terribly concerned about needing to grant deck read access to our testgrid bucket, but I'd also be in favour of being able to just pull it from testgrid directly.

Reply:
Thinking more about it, I'm actually maybe not a fan of giving deck read access to the testgrid bucket: while it's fine _today_, when we move GCS browsing into deck, unless we put in our own access controls that effectively gives the world the same read access it has (at least for directory listing, if not for actually retrieving the files).

I think it might make sense to make the config proto (only) world-readable though — it has a public format based on public data produced by a public tool, and local development without being able to read it is already somewhat annoying when things (i.e. gubernator, perhaps future-deck) want to read it.

Reply:
https://github.com/kubernetes/test-infra/blob/60bca00c14b62d24af333aa3e160dc56368679ac/testgrid/cmd/configurator/main.go#L148 is what uploads the config proto

Reply:
#11318